homeassistant/components/withings/common.py

```diff
@@ -751,19 +752,19 @@ async def async_get_all_data(self) -> dict[MeasureType, Any] | None:
                         for flow in self._hass.config_entries.flow.async_progress_by_handler(
                             const.DOMAIN
                         )
-                        if flow.context == context
+                        if flow["context"].get(const.PROFILE) == self._profile
```

Comment:
Previously this would raise an exception, and with luck the exception would prevent the second authentication flow from being created.